addons/mail/static/tests/chat_window/chat_window_tests.js

```diff
@@ -427,6 +427,36 @@ QUnit.test("open 3 different chat windows: not enough screen width", async (asse
     );
 });
 
+QUnit.test("close hidden chat window and check if there is still the right amount of visible chat windows", async (assert) => {
+    const pyEnv = await startServer();
+    pyEnv["discuss.channel"].create([
+        { name: "Channel_1" },
+        { name: "Channel_2" },
+        { name: "Channel_3" },
+        { name: "Channel_4" },
+        { name: "Channel_5" },
+    ]);
+    patchUiSize({ width: 900 });
+    await start();
+    //open all chat windows
+    await click("button i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem:contains(Channel_1)");
+    await click("button i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem:contains(Channel_2)");
+    await click("button i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem:contains(Channel_3)");
+    await click("button i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem:contains(Channel_4)");
+    await click("button i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem:contains(Channel_5)");
+    assert.containsOnce($, ".o-mail-ChatWindow-header:contains(Channel_1)");
+    assert.containsOnce($, ".o-mail-ChatWindow-header:contains(Channel_5)");
+    await click(".o-mail-ChatWindowHiddenToggler");
+    await click(".o-mail-ChatWindow-header:contains(Channel_2) .o-mail-ChatWindow-command[title='Close Chat Window']");
+    await click(".o-mail-ChatWindowHiddenToggler");
```

Comment:
I would add a comment that this closes the chat window hidden menu.
I wasn't aware that it was kept open, and it seems like a soft spec so could change in the future.
Assertion below on `.o-mail-ChatWindow` relies on it being closed, therefore it's important to clearly understand the intent of this `click()` action